### PR TITLE
wpa_supplicant: Overhaul config

### PR DESCRIPTION
**Pull request by ReillyBrogan**
Changes:
- Rewrite config file to be based off of the upstream defconfig. This makes merging in upstream changes easier
- Enable a large number of options aimed at increasing compatibility with as many networks as reasonably possible. Some of these were upstream defaults that were not enabled as we updated versions.
- Did four-way merge with Arch, Debian, Ubuntu, and Fedora configs to determine what was commonly enabled and modified.
- Added extra bugfix patch preventing potential reconnection after disconnecting from a network
- Changed default TLS ciphers, hopefully fixing the certificate issues once and for all
- Added eapol_test utility for debugging purposes
- Added man pages
- Removed unnecessary systemd services (only DBUS-activation is necessary on Solus)
- Remove old sleep workaround

### CODE REVIEW

**Comment by ReillyBrogan** [comment]
Draft until after sync

**Review by joebonrichie — APPROVED** [review]

**Comment by joebonrichie** [comment]
@EbonJaeger possibly a good idea to mention this in sync notes in case of problems or if it sorts out peeps issues with 2.10.

**Comment by EbonJaeger** [comment]
Definitely. It should also be merged post-sync to get the most testing it can.

**Review by GZGavinZhao — COMMENTED** [review]
Is the `pspec_x86_64.xml` up-to-date? After I built it locally, `git diff` tells me that the file `/usr/lib/systemd/system-sleep/wpa_supplicant.sh` is removed.

**Comment by joebonrichie** [comment]
> Is the `pspec_x86_64.xml` up-to-date? After I built it locally, `git diff` tells me that the file `/usr/lib/systemd/system-sleep/wpa_supplicant.sh` is removed.

Nice catch, it indeed shouldn't be there any longer.

File: packages/w/wpa_supplicant/pspec_x86_64.xml [hunk 1]
```diff
@@ -19,28 +19,33 @@
 </Description>
         <PartOf>network.base</PartOf>
         <Files>
+            <Path fileType="executable">/sbin/eapol_test</Path>
             <Path fileType="executable">/sbin/wpa_cli</Path>
             <Path fileType="executable">/sbin/wpa_passphrase</Path>
             <Path fileType="executable">/sbin/wpa_supplicant</Path>
             <Path fileType="library">/usr/lib/systemd/system-sleep/wpa_supplicant.sh</Path>
```
Comment (joebonrichie) [inline packages/w/wpa_supplicant/pspec_x86_64.xml:26]: Needs rebuilding, so this entry is removed

**Review by joebonrichie — CHANGES_REQUESTED** [review]

**Review by joebonrichie — APPROVED** [review]

**Pull request merged**